services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/schema/SynapseSchemaBootstrapImpl.java

```diff
@@ -114,6 +116,23 @@ void registerSchemaIfDoesNotExist(UserInfo admin, JsonSchema schema) throws Reco
 		request.setSchema(schema);
 		jsonSchemaManager.createJsonSchema(admin, request);
 	}
+	
+	/**
+	 * Use the latest version of each referenced schema.
+	 * @param schema
+	 */
+	void replaceReferencesWithLatestVersion(JsonSchema schema) {
+		for (JsonSchema subSchema : SubSchemaIterable.depthFirstIterable(schema)) {
```

Comment:
Consider adding a method traverse, that takes in input a consumer and/or a filter.